### PR TITLE
Add the "/query/directory" federation API to the client

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: federationclient.go [hunk 1]
```diff
@@ -147,3 +147,14 @@ func (ac *FederationClient) LookupStateIDs(s ServerName, roomID, eventID string)
 	err = ac.doRequest(req, &res)
 	return
 }
+
+// LookupRoomAlias looks up a room alias hosted on the remote server.
+// The domain part of the roomAlias must match the name of the server it is
+// being looked up on.
```
Comment (Kegsay) [inline federationclient.go]: If the room alias does not exist on this server, what happens?
 - An error is returned? What does that look like?
 - A  `RespDirectory` is returned which has `len(servers) == 0`?

**Comment by Kegsay** [comment]
Otherwise LGTM!

File: federationclient.go (hunk 1 above)
Comment (NegativeMjark) [inline federationclient.go]: Commented b7bcebb

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**